bundle/jsky.app.ot/src/main/scala/jsky/app/ot/gemini/parallacticangle/PositionAnglePanel.scala

```diff
@@ -221,15 +217,26 @@ class PositionAnglePanel[I <: SPInstObsComp with PosAngleConstraintAware,
   }
 
   /**
+    * The actual copying of a given pos angle to the data object.
+    * This is done explicitly to avoid overwriting based on minor differences in double precision and to avoid
+    * adding 180 to the position angle, which causes issues.
+    **/
+  private def setInstPosAngle(newAngleDegrees: Double): Unit = editor.foreach { e =>
+    val Precision = 0.005
+    val oldAngleDegrees = e.getDataObject.getPosAngle
+    if (Math.abs(oldAngleDegrees - newAngleDegrees) >= Precision && Math.abs(Math.abs(oldAngleDegrees - newAngleDegrees) - 180) >= Precision) {
+      e.getDataObject.setPosAngle(newAngleDegrees)
```

Comment:
Yes, the PA is still user-editable. We decided at the time of parallactic angle implementation that the user should be able to select parallactic angle mode, have the parallactic angle displayed in the calculation area, but then manually override the PA. I think it was Andy that specifically requested this functionality.

As for the 180 flip, this is just a horror story of BAGS and parallactic angle fighting each other.

A number of things trigger the parallactic angle calculation, which since the target model change, now reconfigures the entire widget with whatever the parallactic angle is calculated to be. Say this is x. Then BAGS will run for x, and find, say, an optimal GS at x+180, setting the PA to x+180. This will trigger a new parallactic angle calculation by the way the code was changed, which will revert the PA to x. Then BAGS does not run, because the hash is still set to PA of x since that was where the last lookup was done. This results in a GS that is then reported out of bounds.